### ISSUE TITLE
Exception for braces on same line

### ISSUE THREAD

**Issue by meherkasam** [nicklockwood/SwiftFormat#637]
Google's style guide states an exception for the K&R-style braces when the opening brace follows a line-wrapped expression. Here's the recommendation:

```
When an open curly brace ({) follows a line-wrapped declaration or expression, it is on the same line as the final continuation line unless that line is indented at +2 from the original line. In that case, the brace is placed on its own line, to avoid the continuation lines from blending visually with the body of the subsequent block.
```

![image](https://user-images.githubusercontent.com/450030/83676081-f86c2f00-a58e-11ea-9039-5da973d4df0a.png)

In the following code sample, I'd hope that running the braces rule wouldn't move the opening brace of the `init` method:

```
class TestClass {
    init(
        foo: Foo
    ) where
        Foo: Fooable,
        Foo.Bar: Barable,
        Foo.Something == Something,
        Foo.SomethingElse == SomethingElse
    {
        self.foo = foo
    }
}
```

Presently, running the following command does shift the opening brace to the right after `== SomethingElse`, making it less readable.

`$ Pods/SwiftFormat/CommandLineTool/swiftformat test.swift --rules braces`

**Comment by meherkasam** [comment]
Link to the Google style guide: https://google.github.io/swift/#line-wrapping

**Comment by meherkasam** [comment]
Thank you for the update. I'm excited for it!

**Comment by nicklockwood** [comment]
@meherkasam fixed in 0.45.1.

**Comment by meherkasam** [comment]
@nicklockwood I tried the braces rule with 0.45.1 and the following was the output:

```
class TestClass {
    init(
        foo: Foo
    ) where
        Foo: Fooable,
        Foo.Bar: Barable,
        Foo.Something == Something,
        Foo.SomethingElse == SomethingElse {
        self.foo = foo
    }
}
```
Is there something else I need to do apart from running the braces rule as follows?

`$ Pods/SwiftFormat/CommandLineTool/swiftformat test.swift --rules braces`

**Comment by nicklockwood** [comment]
@meherkasam you need to enable the `wrapMultilineStatementBraces` rule as well.

**Comment by meherkasam** [comment]
Thank you, that worked!

**Comment by meherkasam** [comment]
@nicklockwood This rule has worked really well for most cases in a sizeable codebase. I did notice one case where it didn't work as expected:
```
class MyClass: NSObject,
    FooProtocol,
    BarProtocol
{
    func someFunction() {}
}
```

I ran the following command:
`$ Pods/SwiftFormat/CommandLineTool/swiftformat test.swift --rules braces,wrapMultilineStatementBraces`

I expected it to remain the same but got the following instead:
```
class MyClass: NSObject,
    FooProtocol,
    BarProtocol {
    func someFunction() {}
}
```

**Comment by nicklockwood** [comment]
@meherkasam thanks for reporting, I'll take a look.

**Comment by nicklockwood** [comment]
@meherkasam fixed in 0.45.2.

**Comment by meherkasam** [comment]
It worked great! Thank you once again.
